### PR TITLE
Feature/client dependency fix

### PR DESCRIPTION
**Pull request by iRhysBradbury**
Client made it into model directory somehow. Removing that Client. Updating client and removing unused plugins in build.
Some strange SBT behaviour fixed.

### CODE REVIEW

File: client/src/main/scala/uk/gov/ons/addressIndex/client/AddressIndexClient.scala [hunk 1]
```diff
@@ -1,76 +1,82 @@
...
+import play.api.libs.ws.{WSClient, WSRequest, WSResponse}
+import uk.gov.ons.addressIndex.client.AddressIndexClientHelper.{AddressIndexServerHost, AddressQuery, UprnQuery}
 
 trait AddressIndexClient {
 
   /**
     * @return a standard web service client
     */
-  def client: WSClient
+  def client : WSClient
```
Comment (Mironor) [inline client/src/main/scala/uk/gov/ons/addressIndex/client/AddressIndexClient.scala]: Why added space ?

**Comment by RichardSmithONS** [comment]
return Future[WSClient] or Future[AddressbySearchResponseContainer] in this version?

File: client/src/test/scala/uk/gov/ons/addressIndex/client/AddressIndexClientTest.scala [hunk 2]
```diff
@@ -1,10 +1,9 @@
 package uk.gov.ons.addressIndex.client
 
-import javax.inject.{Inject, Singleton}
 import org.scalatest.FlatSpec
-import play.api.libs.ws.WSClient
 
 class AddressIndexClientTest extends FlatSpec {
-  ignore "no tests" in {
+  ignore should "no tests" in {
```
Comment (Mironor) [inline client/src/test/scala/uk/gov/ons/addressIndex/client/AddressIndexClientTest.scala]: Can you just explain what is this? (this is not approve-blocking)

File: client/src/main/scala/uk/gov/ons/addressIndex/client/AddressIndexClient.scala [hunk 3]
```diff
@@ -16,61 +16,72 @@ trait AddressIndexClient {
...
-  sealed abstract class AddressIndexPath(val path : String, val method : String)
+  implicit def str2host(h: String): AddressIndexServerHost = AddressIndexServerHost(h)
 
-  case class AddressIndexServerHost(value : String)
+  sealed abstract class AddressIndexPath(val path: String, val method: String)
 
-  implicit class AddressIndexPathToWsAugmenter(h : AddressIndexPath)(implicit client : WSClient) {
-    def toReq() : WSRequest = {
+  implicit class AddressIndexPathToWsAugmenter(p: AddressIndexPath)(implicit client: WSClient, host: AddressIndexServerHost) {
+    def toReq(): WSRequest = {
```
Comment (Mironor) [inline client/src/main/scala/uk/gov/ons/addressIndex/client/AddressIndexClient.scala]: Add this method to AddressIndexPath (implemented, with implicit parameters)
Or leave it be if it doesn't work.

File: client/src/test/scala/uk/gov/ons/addressIndex/client/AddressIndexClientTest.scala (hunk 2 above)
Comment (iRhysBradbury) [inline client/src/test/scala/uk/gov/ons/addressIndex/client/AddressIndexClientTest.scala]: No tests, wanted something to come up when you run sbt test though, telling you theres no tests, hence ignore

File: client/src/main/scala/uk/gov/ons/addressIndex/client/AddressIndexClient.scala (hunk 3 above)
Comment (iRhysBradbury) [inline client/src/main/scala/uk/gov/ons/addressIndex/client/AddressIndexClient.scala]: ? This was the half way mark we just agreed on.

Comment (Mironor) [inline client/src/main/scala/uk/gov/ons/addressIndex/client/AddressIndexClient.scala]: Ah, fine, just wonder what is the reason
Why "upgrade" class from the outside when you can directly add the method to it?
If you wan't to leave it, leave it

**Pull request merged**